File: COM1786-CW1-Project-5/COM1786-CW1-Project-5/app/src/main/java/com/example/comp1786cw1project3/MainActivity.java
package com.example.comp1786cw1project3;

import androidx.fragment.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;

import com.example.comp1786cw1project3.databinding.ActivityMainBinding;
import com.example.comp1786cw1project3.feature.homepage.HomeFragment;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class MainActivity extends BaseActivity {
    @Override
    public Fragment startDestination() {
        return HomeFragment.newInstance();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LayoutInflater inflater = LayoutInflater.from(this);
        ActivityMainBinding viewBinding = ActivityMainBinding.inflate(inflater);
        setContentView(viewBinding.getRoot());
    }
}

File: COM1786-CW2-Project-5/COM1786-CW2-Project-5/app/src/main/java/com/example/comp1786cw2project2/MainActivity.kt
package com.example.comp1786cw2project2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.com1786_cw2project1.R
import com.example.comp1786cw2project2.feature.home.HomeFragment
import com.tunjid.androidx.navigation.stackNavigationController
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    val stackNavigator by stackNavigationController(R.id.navRootHost)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        stackNavigator.push(HomeFragment.newInstance())
    }
}